src/modules/job-ingest/validators/validate-schema.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python
+
+##############################################################
+#  Copyright 2018 Lawrence Livermore National Security, LLC
+#  (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+#
+#  This file is part of the Flux resource manager framework.
+#  For details, see https://github.com/flux-framework.
+#
+#  SPDX-License-Identifier: LGPL-3.0
+##############################################################
+
+import sys
+import argparse
+import json
+import jsonschema
+
+
+def emit(object_dict):
+    s = json.dumps(object_dict, separators=(",", ":"))
+    sys.stdout.write(s + "\n")
+    sys.stdout.flush()
+
+
+def validate(schema, line):
+    try:
+        jobspec = json.loads(line)
+        jsonschema.validate(jobspec, schema)
+    except ValueError as e:
+        return (1, str(e))
+    except jsonschema.exceptions.ValidationError as e:
+        return (1, e.message)
+    return (0, None)
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--schema", "-s", type=str, required=True)
+args = parser.parse_args()
+
+try:
+    with open(args.schema) as fd:
+        schema = json.load(fd)
+except (OSError, IOError) as e:
+    sys.exit(args.schema + ": " + e.strerror)
+except ValueError as e:
+    sys.exit(args.schema + ": " + str(e))
+
+sys.stderr.write("ready\n")
```

Comment:
For printing to stderr in python, I recommend the following:
```python
from __future__ import print_function

print("ready", file=sys.stderr)
```

Reply:
OK, thanks!

Reply:
Is the `from __future__` a python 2/3 compatibility thing?

Reply:
> Is the from __future__ a python 2/3 compatibility thing?

Yep!  In py2, print is a statement, while in py3 it is a function.  The print function is what add the support for the `file` keyword arg. This pulls in the backported print function into py2 (and is a no-op is py3).